src/gsy_e/models/strategy/external_strategies/__init__.py

```diff
@@ -369,8 +369,14 @@ def _offer_aggregator_impl(
                 "area_uuid": self.device.uuid,
                 "market_type": market.type_name,
                 "message": response_message}
-        except (OrderCanNotBePosted, GSyException):
-            logging.exception("Error when handling offer on area %s", self.device.name)
+        except (OrderCanNotBePosted, GSyException) as ex:
+            if isinstance(ex, OrderCanNotBePosted):
+                # This can happen in the normal flow, when the user sends an incorrect offer
+                logging.info("Error when handling offer on area %s. %s", self.device.name, ex)
+            else:
+                # This is more unexpected and we might want to be notified
+                logging.exception("Error when handling offer on area %s. %s", self.device.name, ex)
```

Comment:
nit, maybe we can chain exception handlers
```suggestion
        except OrderCanNotBePosted as ex:
                 # This can happen in the normal flow, when the user sends an incorrect offer
                logging.info("Error when handling offer on area %s. %s", self.device.name, ex)
        except OrderCanNotBePosted as ex:
                 # This is more unexpected and we might want to be notified
                logging.exception("Error when handling offer on area %s. %s", self.device.name, ex)
```

Reply:
The issue with this is that we still need to execute these lines inside the exception: https://github.com/gridsingularity/gsy-e/blob/228599c33ba283021d1357309bec3f71428da739/src/gsy_e/models/strategy/external_strategies/__init__.py#L374-L380
I think that, if we chain exception handlers as you suggest, there's no way to do that without duplicating code. Or do you see an alternative?

Reply:
I see, in this case all good. Thanks!

Reply:
There is also the `finally` statement that could potentially be used here, but frankly this is one of the least frequently used features by me because I find the exception control flow much more confusing than with if / else's. It is feasible, but I would probably discourage you from trying it.

Reply:
I thought about it initially, but the `finally` is executed even when the exceptions are not raised. In our case, however, the code should only be executed when an exception is raised (it should not be executed when everything works smoothly).